### PR TITLE
inkcut: avoid dirty PYTHONPATH

### LINKED ISSUE #159179: inkcut: crashes when picking up invalid libs from PYTHONPATH
### Describe the bug

`inkcut` currently uses Python 3.9.

However, when running it with Python 3.10 libraries on the PYTHONPATH, it will pick those up and crash. It should probably ignore the PYTHONPATH.

### Steps To Reproduce
1. `nix-shell -p inkcut python310Packages.atom`
2. `inkcut`

This gives:

```
Logging to /home/aengelen/.config/inkcut/logs/inkcut.txt
2022-02-11 09:52:40,451 | INFO    | inkcut | ========================================
2022-02-11 09:52:40,451 | INFO    | inkcut | Inkcut launched
2022-02-11 09:52:40,451 | INFO    | inkcut | Version: 2.1.3
2022-02-11 09:52:40,452 | INFO    | inkcut | Python: 3.9.9 (main, Nov 15 2021, 18:05:17) 
[GCC 10.3.0]
2022-02-11 09:52:40,452 | INFO    | inkcut | System: uname_result(system='Linux', node='rigter', release='5.16.3', version='#1-NixOS SMP PREEMPT Thu Jan 27 11:03:05 UTC 2022', machine='x86_64')
2022-02-11 09:52:40,452 | INFO    | inkcut | Executable: /nix/store/x24kxyqwqg2ln8kh9ky342kdcmhbng3h-python3-3.9.9/bin/python3.9
2022-02-11 09:52:40,452 | INFO    | inkcut | Args: ['/nix/store/ydxrj57cldsfmma0s86l57dqlmmzcbn2-inkcut-2.1.3/bin/inkcut']
2022-02-11 09:52:40,469 | INFO    | inkcut | Qt: 5.15.3 Api:
2022-02-11 09:52:40,469 | INFO    | inkcut | ========================================
2022-02-11 09:52:40,473 | ERROR   | inkcut | Traceback (most recent call last):
  File "/nix/store/ydxrj57cldsfmma0s86l57dqlmmzcbn2-inkcut-2.1.3/lib/python3.9/site-packages/inkcut/app.py", line 97, in main
    from inkcut.core.workbench import InkcutWorkbench
  File "/nix/store/ydxrj57cldsfmma0s86l57dqlmmzcbn2-inkcut-2.1.3/lib/python3.9/site-packages/inkcut/core/workbench.py", line 20, in <module>
    from atom.api import Str, Instance
  File "/nix/store/ppk8s3dlsbsqnsl76fblwn0b9b55ph6j-python3.10-atom-0.7.0/lib/python3.10/site-packages/atom/api.py", line 11, in <module>
    from .atom import AtomMeta, Atom, observe, set_default
  File "/nix/store/ppk8s3dlsbsqnsl76fblwn0b9b55ph6j-python3.10-atom-0.7.0/lib/python3.10/site-packages/atom/atom.py", line 13, in <module>
    from .catom import (
ModuleNotFoundError: No module named 'atom.catom'

Traceback (most recent call last):
  File "/nix/store/ydxrj57cldsfmma0s86l57dqlmmzcbn2-inkcut-2.1.3/bin/.inkcut-wrapped", line 9, in <module>
    sys.exit(main())
  File "/nix/store/ydxrj57cldsfmma0s86l57dqlmmzcbn2-inkcut-2.1.3/lib/python3.9/site-packages/inkcut/app.py", line 97, in main
    from inkcut.core.workbench import InkcutWorkbench
  File "/nix/store/ydxrj57cldsfmma0s86l57dqlmmzcbn2-inkcut-2.1.3/lib/python3.9/site-packages/inkcut/core/workbench.py", line 20, in <module>
    from atom.api import Str, Instance
  File "/nix/store/ppk8s3dlsbsqnsl76fblwn0b9b55ph6j-python3.10-atom-0.7.0/lib/python3.10/site-packages/atom/api.py", line 11, in <module>
    from .atom import AtomMeta, Atom, observe, set_default
  File "/nix/store/ppk8s3dlsbsqnsl76fblwn0b9b55ph6j-python3.10-atom-0.7.0/lib/python3.10/site-packages/atom/atom.py", line 13, in <module>
    from .catom import (
ModuleNotFoundError: No module named 'atom.catom'
```

Note the /nix/store/ppk8s3dlsbsqnsl76fblwn0b9b55ph6j-python3.10-atom-0.7.0 in the stack trace: it's likely picking those up from the `PYTHONPATH` of the shell.

### Expected behavior

Inkcut would start, ignoring the PYTHONPATH

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.16.3, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.6.0`
 - channels(root): `"nixos-20.03pre194293.2436c27541b"`
 - nixpkgs: `/home/aengelen/nixpkgs`
```

nixpkgs at 98bb5b77c8c6666824a4c13d23befa1e07210ef1

### PR DESCRIPTION
**Pull request by raboof**
###### Description of changes

This avoids picking up libraries from the users' PYTHONPATH, which
might conflict with the libraries that inkcut actually expects to
run with.

You can reproduce that problem with `nix-shell -p inkcut python310Packages.atom --run inkcut`:
it will pick up the python 3.10 atom while inkcut is still built with pyton 3.9,
leading to an error on startup.

Fixes #159179

I'm not familiar enough with python to say whether this is something that should
be done by default by `buildPythonApplication`, so let's do it just for this
application to begin with.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: doc/contributing/submitting-changes.chapter.md [hunk 1]
```diff
@@ -236,7 +236,7 @@ The `master` branch is the main development branch. It should only see non-break
 
 ### Staging branch {#submitting-changes-staging-branch}
 
-The `staging` branch is a development branch where mass-rebuilds go. It should only see non-breaking mass-rebuild commits. That means it is not to be used for testing, and changes must have been well tested already. If the branch is already in a broken state, please refrain from adding extra new breakages.
+The `staging` branch is a development branch where mass-rebuilds go. Mass rebuilds are commits that cause rebuilds for many packages, like more than 500 (or perhaps, if it's 'light' packages, 1000). It should only see non-breaking mass-rebuild commits. That means it is not to be used for testing, and changes must have been well tested already. If the branch is already in a broken state, please refrain from adding extra new breakages.
```
Comment (Artturin) [inline doc/contributing/submitting-changes.chapter.md:239]: Doesn't belong in this pr, please make a new pr so we can discuss it without blocking this one

Comment (raboof) [reply doc/contributing/submitting-changes.chapter.md:239]: oops, already PR'ed that separately at #172164, accidentally left the commit in this one. will remove.
